### PR TITLE
Don't duplicate stock lines

### LINKED ISSUE #37: Inbound shipment : Duplicated item_lines
Raised by Bijay in https://github.com/openmsupply/openmsupply-client/issues/1206

The problem is that mSupply ends up with duplicated `[item_line]` records when the status changes from `Delivered` to `Verified`

To reproduce:

- From Web client, Click on Replenishment -> Inbound Shipment
- Create a new Inbound Shipment clicking New Shipment
- Add at least one batch line
- Confirm by changing status to 'Delivered'
- Sync and then change status to 'Verified'

The issue is that when the second sync takes place ( update status to verified ) the `item_line` updates have an incorrect `record_ID`. This means that these lines are hit: https://github.com/sussol/msupply/blob/master/Project/Sources/Methods/sync_process_sync_record.4dm#L106-L108
and the status changes from `U` to `I` which creates a new `[item_line]` for each `[stock_line]`

I cannot see where in the remote-server database the `[item_line]ID` is being retained. Perhaps it isn't? Regardless, in the update, only the `[transact]` needs updating, so perhaps limit the update to that table?

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #37 

Simple change to check for an existing ID before making a new one.
and then wandered offroad a little..

## Testing

- [ ] As per issue really.. create an inbound, add a line, set status to delivered then verified. Check the central server: it was duplicating the `[item_line]` record

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,16 @@
 # Open mSupply
 
-### Client [README](client/README.md)
-### Server [README](server/README.md)
\ No newline at end of file
+This is the home of an open source version of the popular [mSupply](https://msupply.org.nz/) - providing pharmaceutical supply chain management and dispensing, all the way from national warehouses to remote clinics, on servers, laptops and mobile. Used in over 30 countries and for over 20 years and now becoming open source.
```
Comment (mark-prins) [inline README.md:3]: I thought that the initial readme was looking a little stark. I've added some words 🤷

File: client/README.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 Welcome! This is the front end application for open mSupply.
 
-This is a multi-lingual, responsive web application making use of the open mSupply [remote-server API](https://github.com/openmsupply/remote-server).
+This is a multi-lingual, responsive web application making use of the open mSupply [remote-server API](https://github.com/open-msupply/server/README.md).
```
Comment (mark-prins) [inline client/README.md:5]: update link..

File: server/service/src/invoice/inbound_shipment/update/generate.rs [hunk 3]
```diff
@@ -97,8 +97,8 @@ pub fn generate_lines_and_stock_lines(
     let mut result = Vec::new();
 
     for invoice_lines in lines.into_iter() {
-        let stock_line_id = uuid();
         let mut line = invoice_lines.clone();
+        let stock_line_id = line.stock_line_id.unwrap_or(uuid());
```
Comment (mark-prins) [inline server/service/src/invoice/inbound_shipment/update/generate.rs:101]: and the actual fix 😁

File: README.md (hunk 1 above)
Comment (clemens-msupply) [reply README.md:3]: btw. I always put one sentence into one line which can make reviewing easier and helps with some diff tools...

File: server/service/src/invoice/inbound_shipment/update/generate.rs (hunk 3 above)
Comment (clemens-msupply) [reply server/service/src/invoice/inbound_shipment/update/generate.rs:101]: this was easy :-)

**Review by clemens-msupply — APPROVED** [review]
Thanks!

**Pull request merged**